@marc-hb sof-kernel-log-check: narrower ignore_str for ICL reboot issue

marc-hb opened the pull request:
2 commits. The main one:

Fixes ac415de ("tools: ignore a false error message with 'panic'")
which was ignoring way too many errors. See previous PR and long source
code comment.

Generally speaking, ignoring errors ("green failures") is extremely
dangerous and should be as narrow as possible.

More specifically here, not even knowing which platforms experience the
issue and what code they print is is really not going to help fix it.

marc-hb commented:
This PR doesn't not add any shellcheck warning in https://travis-ci.org/github/thesofproject/sof-test/builds/725765539, they were all already there.

I didn't to stress test ICL to test this change, #361 was enough. Please help review #361 too.

xiulipan commented:
@marc-hb I will close this PR. The issue can be seen on any device, but the reproduce rate on ICL is very high. So we should ignore the info for all platforms.

marc-hb commented:
> @marc-hb I will close this PR. 

You merged #365 before all discussions were finished and now you're closing this one before any discussion has started.

> but the reproduce rate on ICL is very high. So we should ignore the info for all platforms.

I can't see the how the second sentence relates to the first one.

("very high" doesn't mean much without some numbers)

xiulipan commented:
#365 is only a workaround. I did not think more discussion is needed for that. I think we have plan to use log level to filter the warning and error message to avoid keyword detection.

As the panic log is not platform specific one, I think we should ignore this info message for all platforms.

marc-hb commented:
> As the panic log is not platform specific one, I think we should ignore this info message for all platforms.

I've never seen it outside ICL across many PRs and the description of https://github.com/thesofproject/sof/issues/3395  is very specifically about ICL.

If/when it happens elsewhere then we want to know exactly when and how and not ignore failures "just in case they might happen". This code makes it trivial to add other platforms later, it's designed for that.

aiChaoSONG commented:
@marc-hb It also happened to CML_HEL_RT5682, check the failure on CML_HEL_RT5682 in https://sof-ci.sh.intel.com/#/result/planresultdetail/319

marc-hb commented:
> @marc-hb It also happened to CML_HEL_RT5682, check the failure on CML_HEL_RT5682 in https://sof-ci.sh.intel.com/#/result/planresultdetail/319

Thanks @aiChaoSONG , I just added `cml`. It took 4 additional characters.

I also updated https://github.com/thesofproject/sof/issues/3395

xiulipan commented:
@marc-hb As the panic info may show on any platforms, why we need to narrower the ignore list? I do not want to add a new platform name here once we see it on another platform.

marc-hb commented:
> I do not want to add a new platform name here once we see it on another platform.

People interested in finding and fixing issues are interested in when, where, how and how often they happened. They don't mind adding 4 characters in one file for that sort of information, that's a small price to pay.

marc-hb commented:
> People interested in finding and fixing issues are interested in when, where, how and how often they happened. They don't mind adding 4 characters in one file for that sort of information, that's a small price to pay.

Anyway this is not even the main point of this PR. The main purpose here is not to miss OTHER issues unrelated to https://github.com/thesofproject/sof/issues/3395

ranj063 reviewed (dismissed):
looks good. the commit title only talks of ICL when the patch is for ICL and CML. only minor though

plbossart commented on tools/sof-kernel-log-check.sh:
```diff
@@ -53,12 +53,29 @@ ignore_str="$ignore_str"'|thermal thermal_zone.*: failed to read out thermal zon
-ignore_str="$ignore_str"'|sof-audio-pci 0000:[0-9a-f]{2}:[0-9a-f]{2}\.[0-9a-f]: status = 0x[0-9a-f]{8} panic = 0x[0-9a-f]{8}'
+case "$platform" in
+    icl|cml)
+        ignore_str="$ignore_str"'|sof-audio-pci 0000:00:1f.3: status = 0x[0]{8} panic = 0x[0]{8}'
```
don't you need a \ before the .?

marc-hb replied:
It won't make any difference in this very particular case because the kernel will always print the exact same character in this position. However it is  good reminder for other cases when matching an actual dot matters so I'm adding it.

aiChaoSONG commented:
Merge this for `sof-audio-pci 0000:00:1f.3: status = 0x00000000 panic = 0x00000000` is only seen on ICL and CML.  If we see this on  more platforms, we can add later.
